client-src/elements/form-field-specs.js

```diff
@@ -77,54 +83,101 @@ const OT_MILESTONE_WEBVIEW_RANGE = {
   later: 'ot_milestone_webview_end',
 };
 
-export const OT_SHIPPED_MILESTONE_DESKTOP_RANGE = {
+const OT_ALL_SHIPPED_MILESTONE_DESKTOP_RANGE = {
   earlier: 'ot_milestone_desktop_start',
+  allLater: 'shipped_milestone',
+  warning: 'Origin trial should start before feature is shipped.',
+};
+
+const ALL_OT_SHIPPED_MILESTONE_DESKTOP_RANGE = {
+  allEarlier: 'ot_milestone_desktop_start',
   later: 'shipped_milestone',
-  error: 'Origin trial must start before feature is shipped.',
+  warning: 'All origin trials should start before feature is shipped.',
```

Comment:
Optional nit: when I encountered this warning message filling out a milestone on a shipping stage, the text felt confusing to me, since it was talking about origin trials stages, which I wasn't editing. It might be worthwhile to specifically mention milestones.

e.g. "Origin trials starting milestones should be before feature shipping milestones."

@jrobbins Maybe you have an opinion here

Reply:
There are two separate checks related to OT and shipped milestones.  This one is for the shipped milestone, in which it compares all the OT start milestones with the one shipped milestone you are editing.  It doesn't matter if you are currently editing the OT start milestones, if they have a value from earlier editing.  The other check is for each of the OT start milestones, which compares each one with all the shipped milestones (for the same platform). 

Regarding the message, I was assuming that an origin trial starts with its starting milestone, but maybe that is not quite right?  So your message seems to be equivalent to mine.  I'm open to changing the language to whatever is more appropriate.

I was thinking it would be useful to include the offending milestone numbers, which should be possible with some kind of dynamic creation of tagged template literals (so the `${milestone}` expressions are not evaluated in these constant definitions, but later when the value is available).

Reply:
As a simple improvement, let's just make the warning text include the word "milestones".  That word will help the user find the referenced field by name.

Reply:
I changed it to a more specific variation of Daniel's suggestion: 
`'All origin trials starting milestones should be before feature shipping milestone.'`
and for the OT_ALL_SHIPPED_* cases:
`'Origin trial starting milestone should be before all feature shipping milestones.'`